src/ngraph/runtime/inference_engine/ie_executable.cpp

```diff
@@ -0,0 +1,143 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/runtime/inference_engine/ie_executable.hpp"
+#include "ngraph/opsets/opset.hpp"
+#include "ngraph/op/get_output_element.hpp"
+#include "ngraph/pass/manager.hpp"
+#include "ngraph/pass/opset1_upgrade.hpp"
+#include "ngraph/runtime/inference_engine/ie_tensor_view.hpp"
+
+ngraph::runtime::inference_engine::IE_Executable::IE_Executable(std::shared_ptr<Function> func,
+                                                                std::string _device)
+{
+    auto opset = ngraph::get_opset1();
+    pass::Manager passes;
+    passes.register_pass<pass::Opset1Upgrade>();
+    passes.run_passes(func);
+
+    for (auto& node : func->get_ops())
+    {
+        if (!opset.contains_op_type(node.get()))
+        {
+            if (node->get_type_info() == op::GetOutputElement::type_info)
+            {
+                // IE currently can handle GetOutuputElement op;
+                continue;
+            }
+            else
+            {
+                std::cout << "UNSUPPORTED OP DETECTED: " << node->get_type_info().name << std::endl;
+                THROW_IE_EXCEPTION << "Detected op not belonging to opset1!";
+            }
+        }
+    }
+
+#ifdef NGRAPH_DEBUG_ENABLE
+    std::cout << "Nodes in test: ";
+    for (auto& node : func->get_ops())
+    {
+        std::cout << node << std::endl;
+    }
+    std::cout << std::endl;
+#endif
+
+    network = InferenceEngine::CNNNetwork(func);
+    device = _device == "INFERENCE_ENGINE" ? "CPU" : _device;
+    set_parameters_and_results(*func);
+}
+
+bool ngraph::runtime::inference_engine::IE_Executable::call(
+    const std::vector<std::shared_ptr<runtime::Tensor>>& outputs,
+    const std::vector<std::shared_ptr<runtime::Tensor>>& inputs)
+{
+    InferenceEngine::Core ie;
+
+    //  Loading model to the plugin (BACKEND_NAME)
+    InferenceEngine::ExecutableNetwork exeNetwork = ie.LoadNetwork(network, device);
```

Comment:
ngraph local var names are all snake_case so this should be exec_network, but it is also customary to use the naming convention of the library you are using so, up for discussion. Same with other variables below.